auslib/test/web/test_dockerflow.py

```diff
@@ -27,10 +27,25 @@ def testHeartbeatWithException(self):
             cr.side_effect = Exception("kabom!")
             # Because there's no web server between us and the endpoint, we receive
             # the Exception directly instead of a 500 error
-            self.assertRaises(Exception, self.client.get, "/__heartbeat__")
+            ret = self.client.get("/__heartbeat__")
+            self.assertEqual(ret.status_code, 502)
+            self.assertEqual(ret.data, "Error occured")
             self.assertEqual(cr.call_count, 1)
 
     def testLbHeartbeat(self):
         ret = self.client.get("/__lbheartbeat__")
         self.assertEqual(ret.status_code, 200)
         self.assertEqual(ret.headers["Cache-Control"], "no-cache")
+
+
+class TestHeartbeatExceptionSentry(ClientTestWithErrorHandlers):
```

Comment:
Good job adding this test, but be carefula bout inheriting from ClientTestWithErrorHandlers. Becasue that class has tests of its own, they end up getting re-run as part of this. To avoid this, you need to do one of two things:
- Move ClientTestWithErrorHandlers to a shared base class
- Don't inherit from ClientTestWithErrorHandlers, and copy its setUp to this class.